kernel/src/schema/mod.rs

```diff
@@ -212,33 +216,107 @@ impl StructField {
             .collect()
     }
 
-    /// Applies physical name mappings to this field
+    /// Applies physical name mappings to this field. If the `column_mapping_mode` is
+    /// [`ColumnMappingMode::Id`], then each StructField will have its parquet field id in the
+    /// [`ColumnMetadataKey::ParquetFieldId`] metadata field.
+    ///
     ///
     /// NOTE: Caller affirms that the schema was already validated by
     /// [`crate::table_features::validate_schema_column_mapping`], to ensure that annotations are
     /// always and only present when column mapping mode is enabled.
-    pub fn make_physical(&self) -> Self {
-        struct MakePhysical;
+    pub fn make_physical(&self, column_mapping_mode: ColumnMappingMode) -> Self {
```

Comment:
i saw this is `pub` before but should it be? i guess we expect engines to call this to transform their own schemas? in that case it makes me wonder if we want to expose col mapping more directly or take a snapshot or something?

Reply:
Yeah I don't immediately see why this should be pub. Kernel is the one constructing physical schemas, so maybe we should hide both this and the column mapping mode.

Reply:
Yeah, I suspect this was a mistake when we made it public. Maybe quickly check with @roeap in case he can think of an engine use case, otherwise move to `pub(crate)`